Product Strategy Update (#1650)
FlowFuse/website: src/handbook/product/strategy.md

# Product Strategy

> Our strategy is an indication and statement of intent for the next 2 years, and reflects FlowFuse’s [vision and strategy](..//company/strategy.md). FlowFuse will allow everyone to build custom applications to achieve their business goals and depend less on off-the-shelf software.  It is a living strategy, so we will revisit on a regular basis and iterate when there is reason to do so; timing is everything. 

Reviewer: ```suggestion
Our strategy is an indication and statement of intent for the next 2 years, and reflects FlowFuse’s [vision and strategy](/handbook/company/strategy.md). FlowFuse will allow everyone to build custom applications to achieve their business goals and depend less on off-the-shelf point solutions. 

It strategy is ever-changing, it will be revised on a regular basis and we will iterate when there is reason to do so; timing is everything. 
```

## Market Vision

> **Our strategic goal is to be the leading "Low-Code IIoT Application Platform"**

Author: IIoT encompasses manufacturing but is not limited to it. IIoT includes:
- oil and gas industry
- supply chain use cases
- agriculture industry
- Utilities

> An IIoT platform is a comprehensive solution that securely collects, integrates, and manages vast volumes of data from IoT endpoints. It's the bridge between Information Technology (IT) and Operational Technology (OT), bringing about a seamless fusion of diverse protocols and data formats, facilitating edge and cloud analytics, and enhancing overall asset management. This digital conduit not only augments traditional OT functionalities but also propels sectors like manufacturing and energy towards a more digitized, efficient future. A robust IIoT platform encapsulates essential elements such as unwavering security, sophisticated automation, and a commitment to sustainability. It fosters an environment where interoperability with a myriad of business applications is not just a feature but a foundational aspect. From meticulous device management and seamless integration to advanced data analytics and application development (see also [IIoT platform definition](https://www.gartner.com/doc/reprints?id=1-2C757S9J&ct=230105&st=sb)).

Reviewer: My take:
- It's harder to innovate at the pace required to extract ROI
- Development of custom solutions is expensive, because 1) it requires lot of people to be involved with different skill sets 2) hard to capture data from the different sources and make it accessable for the org
   - And so traditional industries select point solutions, leaving ROI on the table
- If all data, regardless of where it was produced or needed to be comsumed, was available and useful for _all_ engineers in the org (mechenical, elektrical) increased ROI solutions can be build and maintained by mid to large sized companies in traditional industries.

> An IIoT platform is a comprehensive solution that securely collects, integrates, and manages vast volumes of data from IoT endpoints. It's the bridge between Information Technology (IT) and Operational Technology (OT), bringing about a seamless fusion of diverse protocols and data formats, facilitating edge and cloud analytics, and enhancing overall asset management. This digital conduit not only augments traditional OT functionalities but also propels sectors like manufacturing and energy towards a more digitized, efficient future. A robust IIoT platform encapsulates essential elements such as unwavering security, sophisticated automation, and a commitment to sustainability. It fosters an environment where interoperability with a myriad of business applications is not just a feature but a foundational aspect. From meticulous device management and seamless integration to advanced data analytics and application development (see also [IIoT platform definition](https://www.gartner.com/doc/reprints?id=1-2C757S9J&ct=230105&st=sb)).

Reviewer: @knolleary I appreciate your thoughts here, I'm still not sure if we should want to be an IIoT Platform, or set another target on the horizon?

> An IIoT platform is a comprehensive solution that securely collects, integrates, and manages vast volumes of data from IoT endpoints. It's the bridge between Information Technology (IT) and Operational Technology (OT), bringing about a seamless fusion of diverse protocols and data formats, facilitating edge and cloud analytics, and enhancing overall asset management. This digital conduit not only augments traditional OT functionalities but also propels sectors like manufacturing and energy towards a more digitized, efficient future. A robust IIoT platform encapsulates essential elements such as unwavering security, sophisticated automation, and a commitment to sustainability. It fosters an environment where interoperability with a myriad of business applications is not just a feature but a foundational aspect. From meticulous device management and seamless integration to advanced data analytics and application development (see also [IIoT platform definition](https://www.gartner.com/doc/reprints?id=1-2C757S9J&ct=230105&st=sb)).

Reviewer 2: (The gartner link in the text has expired)

I've been trying to come up with a response that helps move this conversation forward... and struggling to find the right words, so this may be a bit rambling and perhaps simplistic.

This links to our [Company Strategy](https://flowfuse.com/handbook/company/strategy/) but I don't see it reference any specific points of the strategy. That makes it hard to see how the two relate.

The company strategy lists the following goals:

 - Drive the growth and success of the Node-RED project
 - Become the standard way of running any Node-RED for professional usage
 - Drive adoption of Node-RED in traditional industries

From a product strategy point of view, the question we have to answer is how each item in this doc relates back to those goals.

The same company strategy leans into the association with Node-RED - we don't currently justify the value Node-RED can bring, we justify the value FlowFuse can bring to existing Node-RED users. A basic question is then, does our vision (for the next 2 years) look beyond that, or is that implicitly what is at the heart of the vision.

We want FlowFuse to be the obvious choice when working with Node-RED in the enterprise.

My hesitation to setting our product strategy to be an IIoT Platform is whether that goes beyond the Node-RED-focus of our existing strategy. But then, I don't have a good label for it if we don't call it a low-code IIoT Platform.

The more I try to get this into words, the more I think we need to just set aside a couple hours and discuss this in real-time.

> An IIoT platform is a comprehensive solution that securely collects, integrates, and manages vast volumes of data from IoT endpoints. It's the bridge between Information Technology (IT) and Operational Technology (OT), bringing about a seamless fusion of diverse protocols and data formats, facilitating edge and cloud analytics, and enhancing overall asset management. This digital conduit not only augments traditional OT functionalities but also propels sectors like manufacturing and energy towards a more digitized, efficient future. A robust IIoT platform encapsulates essential elements such as unwavering security, sophisticated automation, and a commitment to sustainability. It fosters an environment where interoperability with a myriad of business applications is not just a feature but a foundational aspect. From meticulous device management and seamless integration to advanced data analytics and application development (see also [IIoT platform definition](https://www.gartner.com/doc/reprints?id=1-2C757S9J&ct=230105&st=sb)).

Author: As discussed in our meeting: 
[Update strategy.md](https://github.com/FlowFuse/website/pull/1650/commits/f9e17c44551445ff782c77de5baba911b767e4af)

Click [here](./verticals.md#industrial-iot-platforms) for more details about the "Low-Code IIoT Application Platforms" market.

### Market Demand & Positioning

> With its prowess in connecting and processing data from IIoT devices, Node-RED and FlowFuse find a fitting place in the sphere of IIoT platforms, becoming an integral tool in industries like manufacturing where the utility of such data is paramount for fostering operational efficacy and stimulating innovation. We exist to empower bottom-up innovation. We allow companies to professionalize their workflows to business critical applications.

Reviewer: It seems like what you're saying is; people need to start automating with Node-RED. Their first step in adoption is just a workflow. (Another way to describe data-capture and transformation, I like it)

From that moment onwards they will understand they need to elevate their workflows to applications. Which is where we come in.

No need to mention the features for Node-RED yet, it's clear from the problem statement in the vision.

- [Implement "Organization" as an additional hierarchy level](https://github.com/FlowFuse/flowfuse/issues/2338): FlowFuse is expanding its enterprise capabilities by introducing "Organization" as an additional hierarchy level, particularly within the FlowFuse Cloud environment. This development is a direct response to the growing demand from enterprise customers utilizing our managed SaaS offering. As businesses increasingly seek efficient ways to manage multiple teams on the FlowFuse Cloud, the implementation of Organizations is set to provide a more structured and seamless management and billing system. This enhancement will not only cater to the needs of larger corporations but also streamline administrative processes, thereby offering a more cohesive and efficient platform for enterprise-level operations.

> - [File Storgae for Instances](https://github.com/FlowFuse/flowfuse/issues/3056) This feature aims to enable users to store data within FlowFuse-hosted Node-RED instances, a requirement for numerous use cases, such as displaying images in dashboards. More importantly, it is essential for most AI models, which need persistent file storage. The introduction of file storage for Node-RED hosted instances is a critical step towards demonstrating Node-RED's capabilities as an enabler for AI applications, furthering its utility and applicability in advanced, data-driven environments.

Reviewer: I don't get what you're saying at all. There's file storage available, and FlowFuse already posted a blog on how to host images for dashboards.

Author: You can add images to your Dashboard if the images are hosted somewhere else that the Node-RED Dashboard can access. 

For example, uploading a TensorFlow model and using it in FF - Node-RED is not possible.

> ### Node-RED Deployments at the Edge (Device Management)

Reviewer: This is the feature, not what the user is looking for. **Why** does the user need this to be happening?

Reviewer: ```suggestion
### Applications on the edge 
```

> Many organizations position Node-RED instances on remote servers, edge or industrial devices, to meet network, infrastructure, or use case requirements. Therefore, management of remote instances is crucial for the overall success of closing the gap between IT and OT. 

Reviewer: Closing the IT/OT gap is a consequence of the work we're doing, not the first goal. The goal is to empower everyone to build applications.

```suggestion
Many organizations position Node-RED instances on remote servers, edge-, or industrial-devices. This practise allows them to meet network-, infrastructure-, or use-case- requirements.
```

FlowFuse already supports this process by creating snapshots on Node-RED instances that can be deployed to multiple remote targets. We will continue to expand the functionalities to allow users to manage existing Node-RED instances, improve their development practices to accelerate development and ensure a reliable and repeatable process, providing more flexible remote deployment options and offering the best support for devices in closed and segmented networks as we are facing it in production environments.

> - [Enhanced Device Settings](https://github.com/FlowFuse/flowfuse/issues/3172): The initiative focuses on expanding the configuration options for Devices, aligning with existing capabilities on hosted Node-RED setups. It aims to empower users to remotely manage their device settings, enhancing the efficiency and flexibility of device fleet management. 

Reviewer: I think what this is really doing is making the edge location a first class citizen. Now it's harder to develop and deploy on the edge compared to a cloud Node-RED. In the next 6 months you want to close this gap.

- FlowFuse Dashboard feature parity with D1.0. A logical step to replace D1.0 .

- Manufacturing Widgets Extension for Dashboard 2.0. This extension should provide very classic manufacturing visualization, making it even easier to create Node-RED Dashboards for manufacturing.

### User-Friendly Low-Code Approach

FlowFuse is built upon the strength of Node-RED's user-friendly, low-code approach, positioning us well within the IIoT market. We believe Node-RED provides the best foundation, but we're committed to not only utilizing its existing capabilities but also contributing to its upstream improvement. By refining Node-RED and adding enhancements, we aim to create an even more intuitive, powerful, and accessible platform. This will enable end-users to swiftly and intuitively build applications, reducing the dependency on expert programmers.

> - [Immersive Node-RED experience](https://github.com/FlowFuse/flowfuse/issues/2246), a separation between the Node-RED Editor and FlowFuse leads to unused features from a UX standpoint in FlowFuse, due to the need to jump between both entities. An immersive Node-RED experience embeds Node-RED into FlowFuse, allowing seamless usage of Node-RED and FlowFuse features.

Reviewer: I'd argue this part of closing the gap between the edge and cloud. What do you think?

Author: Two sides of the same coin: UX and Functionality

Reviewer: If it is, let's merge it and remove the section "User-Friendly Low-Code Approach"
